src/pudl/transform/eia.py

```diff
@@ -1149,133 +1140,171 @@ def fix_balancing_authority_codes_with_state(
-            "clean_generation_fuel_nuclear_eia923",
             "clean_generators_eia860",
-            "clean_ownership_eia860",
-            "clean_plants_eia860",
-            "clean_utilities_eia860",
+            "clean_boiler_fuel_eia923",
         ]
-        # This would be better, but currently results in a circular import error due to
-        # the use of occurrence_consistency in pudl.output.eia860. Need to untangle.
-        # asset_key.to_python_identifier(): AssetIn()
-        # for eia_asset in load_assets_from_modules(
-        #     [pudl.transform.eia860, pudl.transform.eia923]
-        # )
-        # for asset_key in eia_asset.keys
-    },
-    outs={
-        table_name: AssetOut(io_manager_key="pudl_sqlite_io_manager")
-        for table_name in sorted(
-            Package.get_etl_group_tables("entity_eia")
-            + Package.get_etl_group_tables("eia860")
-            + Package.get_etl_group_tables("eia923")
-        )
     },
     config_schema={
         "debug": Field(
             bool,
             default_value=False,
             description=(
-                "If True, informational columns will be added into "
-                "boiler_generator_assn"
+                "If True, debugging columns will be added to boiler_generator_assn."
             ),
         ),
     },
     required_resource_keys={"dataset_settings"},
+    io_manager_key="pudl_sqlite_io_manager",
 )
-def eia_transform(context, **eia_transformed_dfs):
-    """Creates DataFrames for EIA Entity tables and modifies EIA tables.
-
-    This function coordinates two main actions: generating the entity tables
-    via ``harvesting()`` and generating the boiler generator associations via
-    ``_boiler_generator_assn()``.
-
-    There is also some removal of tables that are no longer needed after the
-    entity harvesting is finished.
-
-    Args:
-        eia_transformed_dfs (dict): a dictionary of table names (keys) and
-            transformed dataframes (values).
-
-    Returns:
-        Dataframes of entity tables transformed EIA tables.
-    """
+def boiler_generator_assn_eia860(context, **eia_transformed_dfs) -> pd.DataFrame:
+    """Create boiler generator association table."""
+    eia_settings = context.resources.dataset_settings.eia
     # Do some final cleanup and assign appropriate types:
     eia_transformed_dfs = {
         name: convert_cols_dtypes(df, data_source="eia")
         for name, df in eia_transformed_dfs.items()
     }
-
-    # Remove the clean_ prefix from the table names.
+    # Remove the clean_ prefix from the table names... temporary compatibility
     eia_transformed_dfs = {
         table_name.replace("clean_", ""): df
         for table_name, df in eia_transformed_dfs.items()
     }
+    return _boiler_generator_assn(
+        eia_transformed_dfs,
+        eia923_years=eia_settings.eia923.years,
+        eia860_years=eia_settings.eia860.years,
+        debug=context.op_config["debug"],
+    )
```

Comment:
Would it be better to just do all this work in one function rather than have this wrapper? I don't think we call `_boiler_generator_assn_eia860` anywhere else.

The `clean_` rename feels janky and I want to modify the underlying function to just use whatever the DF names are as-is.

Would also be simpler to just let the underlying function depend directly on `context` and pull out the EIA settings it needs, rather than breaking them out individually here I think.

Reply:
Removed the compostable wrapper.